index.bs

```diff
@@ -232,27 +232,137 @@ Non-Functional Requirements {#requirements-non-functional}
+[DNS-SD](https://tools.ietf.org/html/rfc6763) over
+[mDNS](https://tools.ietf.org/html/rfc6762).  To do so, endpoints must
+use the service name "_openscreen._udp.local".
+
+Advertising endpoints must include DNS TXT records with the following
+keys and values:
+
+- key "fp" with value of the certificate fingerpint of the advertising endpoint.
+  The format of the fingerprint is defined by [RFC 8122 section
+  5](https://tools.ietf.org/html/rfc8122#section-5), excluding the
+  "fingerprint:" prefix and including the hash function, space, and hex-encoded
+  fingerprint.  The fingerprint value also functions as an ID for the endpoint.
+  All endpoints must support the following hash functions: "sha-256", "sha-512".
+  Endpoints must not support the following hash functions: "md2", "md5". 
+
+- key "time" with arbitrary unsigned integer value.  To indicate that
+  metadata has changed, the advertising endpoint must update it to a
+  greater value.  This signals to the discovering endpoint that it
+  should connect to the advertising endpoint to discover updated
+  metadata.
+ 
+An endpoint that advertises this service name must implement the
+transport and metadata discovery mechanism with QUIC described below.
+If a future version of this protocol uses mDNS but breaks
+compatibility with that mechanism, it should change the service name
+to a new value indicating a new mechanism for metadata discovery
+mechanism.  If future versions of this protocol or extensions to this
+protocol are compatible with the metadata discovery mechanism, they
+can use it to indicate support for future versions and extensions.
+
+
+Transport and metadata discovery with QUIC {#transport}
+=======================================================
+
+If a discovering endpoint wants to connect to an advertising endpoint, or to
+learn further metdata about it, it initiates a
+[QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-16) connection to
+the IP and port from the SRV record.  Prior to authentication, some message may be
+exchanged (such as further metadata), but such info should be treated as
+unauthenticated (such as indicating to a user that a display name of an
+unauthenticated endpoint is unauthenticated).  
+
+To learn further metadata, an endpoint may send an endpoint-info-request
+message (see Appendix A) and receive back an endpoint-info-response message.  The
+messages may contain the following information with the following meaning:
+
+- display-name: The display name of the responding endpoint intended to be
+  displayed to a user by the requesting endpoint.  If the responding endpoint is
+  not yet authenticated, the requesting endpoint should make UI affordance for
+  indicating to the user that the display name is not yet authenticated.  If the
+  responding endpoint changes its display name, the requesting endpoint should
+  make UI affordance for indicating to the user that the display name has
+  changed. 
```

Comment:
To change a display name, an advertising agent (I'll go with that term) would need to change the instance name, the TXT "mv" (I'll go with that term).  A listening agent would then request the new name to verify it.  But it could do that over the existing QUIC connection.  It wouldn't need to drop the existing QUIC connection.

Reply:
I suppose we could optimize this by just allowing the adveritising agent to push new messages (events) that such info had been updated without the listening agent requesting it.